Sources/XCTest/XCTestCase.swift

```diff
@@ -308,4 +308,62 @@ extension XCTestCase {
             completionHandler(error)
         }
     }
+    
+    /// Creates and returns an expectation for a notification.
+    ///
+    /// - Parameter notificationName: The name of the notification the
+    ///   expectation observes for.
+    /// - Parameter object: The object whose notifications the expectation will
+    ///   receive; that is, only notifications with this object are observed by
+    ///   the test case. If you pass nil, the expectation doesn't use
+    ///   a notification's object to decide whether it is fulfilled.
+    /// - Parameter handler: If provided, the handler will be invoked when the
+    ///   notification is observed. It will not be invoked on timeout. Use the
+    ///   handler to further investigate if the notification fulfills the 
+    ///   expectation.
+    ///
+    /// - Note: The function used to observe the notification 
+    ///   (addObserverForName(_ name: String?, object obj: AnyObject?, 
+    ///   queue queue: NSOperationQueue?, 
+    ///   usingBlock block: (NSNotification) -> Void) -> NSObjectProtocol) 
+    ///   allows a nil notification name for its criteria of observing
+    ///   notifications, the Objective-C implementation of this function 
+    ///   requires a notification name.
+    public func expectationForNotification(notificationName: String, object: AnyObject?, handler: XCNotificationExpectationHandler?) -> XCTestExpectation {
```

Comment:
Super minor nit-pick: Technically the Apple XCTest signature for this method is:

```swift
public func expectationForNotification(notificationName: String, object objectToObserve: AnyObject?, handler: XCNotificationExpectationHandler?) -> XCTestExpectation
```

I don't think the parameter name is important, but I'd personally prefer to be 100% consistent. :stuck_out_tongue: